src/main/java/it/gov/pagopa/pu/migration/connector/fileshare/FileShareServiceImpl.java

```diff
@@ -16,7 +18,17 @@ public FileShareServiceImpl(FileShareClient client) {
 
   @Override
   public Long uploadIngestionFlowFile(Long organizationId, IngestionFlowFileType ingestionFlowFileType, Resource file, String accessToken) {
-    return client.uploadIngestionFlowFile(organizationId, ingestionFlowFileType, file, accessToken);
+    try {
+      return client.uploadIngestionFlowFile(organizationId, ingestionFlowFileType, file, accessToken);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.BAD_REQUEST) {
+        e.getResponseBodyAsString();
+        if (e.getResponseBodyAsString().contains("\"code\":\"INVALID_FILE\"")) {
+          throw new IllegalArgumentException("File name must contain a valid version: [1_0, 1_1, 1_2, 1_3]");
```

Comment:
The 400 INVALID_FILE could be for other reasons... you should set as message the error description of the error body returned...